resources/waterheater.rb

```diff
@@ -1199,11 +1231,19 @@ def self.create_new_schedule_manager(t_set, model, wh_type)
     OpenStudio::Model::SetpointManagerScheduled.new(model, new_schedule)
   end
 
-  def self.create_new_heater(name, cap, fuel, vol, ef, re, jacket_r, t_set, space, oncycle_p, offcycle_p, ec_adj, wh_type, cyc_derate, nbeds, model, runner)
+  def self.create_new_heater(name, cap, fuel, vol, ef, re, jacket_r, t_set, space, oncycle_p, offcycle_p, ec_adj, wh_type, cyc_derate, nbeds, model, runner, ua)
     new_heater = OpenStudio::Model::WaterHeaterMixed.new(model)
     new_heater.setName(name)
-    act_vol = calc_actual_tankvol(vol, fuel, wh_type)
-    u, ua, eta_c = calc_tank_UA(act_vol, fuel, ef, re, cap, wh_type, cyc_derate, jacket_r, runner)
+    if ua.nil?
+      act_vol = calc_actual_tankvol(vol, fuel, wh_type)
+      u, ua, eta_c = calc_tank_UA(act_vol, fuel, ef, re, cap, wh_type, cyc_derate, jacket_r, runner)
+      new_heater.setHeaterThermalEfficiency(eta_c)
+      new_heater.setHeaterFuelType(HelperMethods.eplus_fuel_map(fuel))
+    else
+      # indirect water heater has standby loss for ua calculation
+      # actual volume is already calculated
+      act_vol = vol
```

Comment:
This seems very complicated that now the `vol` passed in is sometimes nominal and sometimes actual. It would be simpler to rename the `vol` variable in this method to `act_vol` and always pass in the actual volume (and move the `calc_actual_tankvol` calls outside this method).

Reply:
Thanks @shorowit .
Good idea of removing calc_ua and calc_vol outside! In this way, create_new_tanks should be more general for being called by different types of water heating systems. I did have a hard time trying to call create_new_heater function for indirect wh. Sorry for ending up with so many if/else statements. Will simplify it!